src/DependencyInjection/Compiler/ExtensionCompilerPass.php

```diff
@@ -156,19 +156,29 @@ private function getManagedClass(Definition $admin, ContainerBuilder $container)
     }
 
     /**
-     * @param array<string, array<string, array<string, string>|bool>> $config
+     * @param array<string, array<string, array<string, string>|int|bool>> $config
      *
-     * @return array<string, array<string, array<string, array<string, array<string, string>>>>> an array with the following structure.
+     * @return array<string, array<string, array<string, array<string, int>>>> an array with the following structure
      *
-     * [
-     *     'global'     => ['<admin_id>'  => ['<extension_id>' => ['priority' => <int>]]],
-     *     'excludes'   => ['<admin_id>'  => ['<extension_id>' => ['priority' => <int>]]],
-     *     'admins'     => ['<admin_id>'  => ['<extension_id>' => ['priority' => <int>]]],
-     *     'implements' => ['<interface>' => ['<extension_id>' => ['priority' => <int>]]],
-     *     'extends'    => ['<class>'     => ['<extension_id>' => ['priority' => <int>]]],
-     *     'instanceof' => ['<class>'     => ['<extension_id>' => ['priority' => <int>]]],
-     *     'uses'       => ['<trait>'     => ['<extension_id>' => ['priority' => <int>]]],
-     * ]
+     * @phpstan-param array<string, array{
+     *     global: bool,
+     *     excludes: array<string, string>,
+     *     admins: array<string, string>,
+     *     implements: array<string, string>,
+     *     extends: array<string, string>,
+     *     instanceof: array<string, string>,
+     *     uses: array<string, string>,
+     *     priority: int,
+     * }> $config
+     * @phpstan-return array{
+     *     global: array<string, array<string, array{priority: int}>>,
+     *     excludes: array<string, array<string, array{priority: int}>>,
+     *     admins: array<string, array<string, array{priority: int}>>,
+     *     implements: array<string, array<string, array{priority: int}>>,
+     *     extends: array<string, array<string, array{priority: int}>>,
+     *     instanceof: array<string, array<string, array{priority: int}>>,
+     *     uses: array<string, array<string, array{priority: int}>>,
```

Comment:
Can we use here `class-string` as key? (based on the existing phpdoc)

Reply:
Done